templates_raw/root/js/feel.js

```diff
@@ -83,38 +83,43 @@
 	np._options = {
 		  interval : 10
 		, playInBackground : false
-		, nsfw : false // not implemented yet
 		};
 
-	np.__timeout = 0;
+	np.__intervall = 0;
+
+	np._setTimer = function(){
+
+        log("Timer refresh called"); // @stripOnBuild
+
+        if(np.__intervall)
+        {
+            window.clearInterval(np.__intervall);
+            log("Timer Cleared"); // @stripOnBuild
+        }
+
+        if(this._state == 0) {
+            np.__intervall = window.setInterval(function () {
+                np._fetch();
+            }, np._options.interval * 1000);
+            log("Timer Set"); // @stripOnBuild
+        }
+	};
 
 	addEvent(np._fetchRequest, "readystatechange", function ()
 	{
 		var req = this;
 		log("XHR onreadystatechange", req.readyState); // @stripOnBuild
-		if ( req.readyState == 4 && req.status == 200 )
-		{
-			var imageURI = req.responseText;
-			if ( imageURI )
-			{
-				var src = imageURI.split('#', 2)
-				  , uri = src[0]
-				  , crawler = (""+ src[1]).toLowerCase();
-				np._pushImage(uri, crawler, function (added)
-				{
-					if ( added && ! np.__timeout )
-					{
-						np.__timeout = window.setTimeout(function ()
-						{
-							np.__timeout = 0;
-							np._fetch();
-						}, np._options.interval * 1000);
-					}
-				});
-			}
-		}
-	});
+		if ( req.readyState == 4 && req.status == 200) {
+				var imageURI = req.responseText;
+				if (imageURI) {
+					var src = imageURI.split('#', 2)
+						, uri = src[0]
+						, crawler = ("" + src[1]).toLowerCase();
+					np._pushImage(uri, crawler);
+				}
+        }
```

Comment:
Changed from time-out to interval. 
In the old implementation, when one Request fails, the time-out is not set again so nichtparasoup hangs.
With the interval, every $interval seconds a fetch is called, regardless what the last one did. 
One advantage of the interval is the robustness, one drawback is, that if images take longer to load, the time between them appearing changes, so nichtparasoup is "hopping"...